pkg/scalers/metrics_api_scaler.go

```
@@ -408,8 +455,147 @@ func getValueFromYAMLResponse(body []byte, valueLocation string) (float64, error
+		serviceEndpoints := &corev1.Endpoints{}
+
+		err := s.kubeClient.Get(ctx, client.ObjectKey{
+			Namespace: namespace,
+			Name:      serviceName,
+		}, serviceEndpoints)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, subset := range serviceEndpoints.Subsets {
+			foundPort := ""
+			for _, port := range subset.Ports {
+				if strconv.Itoa(int(port.Port)) == podPort {
+					foundPort = fmt.Sprintf(":%d", port.Port)
+					break
+				}
+			}
+			if foundPort == "" {
+				s.logger.Info(fmt.Sprintf("Warning : could not find port %s in endpoint slice for service %s.%s definition. Will infer port from %s scheme", podPort, serviceName, namespace, url.Scheme))
+			}
+			for _, address := range subset.Addresses {
+				if address.NodeName != nil {
+					endpointUrls = append(endpointUrls, fmt.Sprintf("%s://%s%s%s", url.Scheme, address.IP, foundPort, url.Path))
+				}
+			}
+		}
+	}
+	return endpointUrls, err
+}
+
 func (s *metricsAPIScaler) getMetricValue(ctx context.Context) (float64, error) {
-	request, err := getMetricAPIServerRequest(ctx, s.metadata)
+	// if we wish to aggregate metric from a kubernetes service then we need to query each endpoint behind the service
+	if s.metadata.aggregateFromKubeServiceEndpoints {
+		endpointsUrls, err := s.getEndpointsUrlsFromServiceURL(ctx, s.metadata.url)
+		if err != nil {
+			s.logger.Error(err, "Failed to get kubernetes endpoints urls from configured service URL. Falling back to querying url configured in metadata")
+		} else {
+			if len(endpointsUrls) == 0 {
+				s.logger.Error(err, "No endpoints URLs were given for the service name. Falling back to querying url configured in metadata")
+			} else {
+				aggregatedMetric, err := s.aggregateMetricsFromMultipleEndpoints(ctx, endpointsUrls)
+				if err != nil {
+					s.logger.Error(err, "No aggregated metrics could be computed from service endpoints. Falling back to querying url configured in metadata")
+				} else {
+					return aggregatedMetric, err
+				}
+			}
+		}
```

I'm not sure if we should fallback here, if a user wants to use the feature and it fails, the fallback can produce unexpected scaling behaviors

---

yes, I was not really sure about it either, but thought that worst case scenario it would at least behave like the current version of metrics API scaler does and just query the metrics API URL

Would you like me to just return error here instead?

---

I'd return an error or, at least, returning an error as default and allowing customizing this via parameter.
I imagine the case where I have 5 instances returning 1000, and suddenly I lose communication with 3 of them, scaling based on 2000 instead of 5000 can produce an unintended scaling in with a service disruption.

In this case, falling back sounds as default risky to me

---

ok @JorTurFer  I [added this commit](https://github.com/kedacore/keda/pull/6565/commits/25f9b0dda7c553634b22780fff4d89896badb725) in order not to fallback & simply return error whenever encountered

---

Hello! 
Seems E2E ran fine again. @JorTurFer did you have time to check the updated version fixed according to your remarks by any chance?
thanks! :)